onnxruntime/core/mlas/inc/mlas.h

```diff
@@ -147,21 +147,25 @@ MlasGemm(
     MLAS_THREADPOOL* ThreadPool
     );
 
-template<typename AType, typename BType>
+struct MLAS_GEMM_U8X8_PARAMETERS {
```

Comment:
>MLAS_GEMM_U8X8_PARAMETERS [](start = 7, length = 25)

so this API supports both uint8 and int8, even though the type of B is uint8_t*?

Reply:
The very first thing either of these templates did was package the parameters into a context structure and then call down into the same kernel. I was moving that dispatching out a level so that then ops could be combined. However, I'm not in love with how that code is looking now. I'm abandoning this PR and doing just some of the refactoring in a new PR.